src/deploy/functions/prepare.ts

```diff
@@ -87,5 +89,28 @@ export async function prepare(context: any, options: any, payload: any): Promise
   const localFnsInRelease = payload.functions.triggers.filter((fn: CloudFunctionTrigger) => {
     return functionMatchesAnyGroup(fn.name, context.filters);
   });
+
+  const res = await gcp.cloudfunctions.listAllFunctions(context.projectId);
+  if (res.unreachable) {
+    const regionsInDeployment = Object.keys(payload.functions.byRegion);
+    const unreachableRegionsInDeployment = regionsInDeployment.filter((region) =>
+      res.unreachable.includes(region)
+    );
+    if (unreachableRegionsInDeployment) {
+      throw new FirebaseError(
+        "The following Cloud Functions regions are currently unreachable:\n" +
+          unreachableRegionsInDeployment.join("\n") +
+          "This deployment contains functions in those regions. Please try again in a few minutes, or exclude these regions from your deployment."
```

Comment:
nit* I think we might want to add a space or a new line before the last sentence for readability. Similar changes to other log lines.

Reply:
Ahhh, good point. Will do